docs/src/proposals/partitioned-inflationary-rewards-distribution.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Partitioned Inflationary Rewards Distribution
+---
+
+## Problem
+
+With the increase of number of stake accounts, computing and redeeming the stake
+rewards at the start block of the epoch boundary becomes very expensive.
+Currently, with 550K stake accounts, the stake reward time has already taken
+more than 10 seconds. This prolonged computation slows down the network, and can
+cause large number of forks at the epoch boundary, which makes the matter even
+worse.
+
+## Proposed Solutions
+
+Instead of computing and reward stake accounts at epoch boundary, we will
+decouple reward computation and reward credit into two phases.
+
+A separate service, "epoch calc service" will be created. The service will
+listen to a channel for any incoming rewards calculation requests, and perform
+the calculation for the rewards. For each block that cross the epoch boundary,
+the bank will send a request to the `epoch calc service`. This marks the start
+of the reward computation phase.
+
+N-1 -- N -- N+1
+     \
+      \
+        N+2
+
+In the above example, N is the start of the new epoch. Two rewards calculation
+requests will be sent out at slot N and slot N+2. To avoid repeated computation
```

Comment:
nit: expand to "Two rewards calculation
requests will be sent out at slot N and slot N+2 because they both cross the epoch boundary and are on different forks"